utils/fixed_decimal/src/decimal.rs

```diff
@@ -388,28 +400,40 @@ impl FixedDecimal {
     /// let mut dec = FixedDecimal::from(42);
     /// assert_eq!("42", dec.to_string());
     ///
-    /// dec.multiply_pow10(3).expect("Bounds are small");
+    /// dec.multiply_pow10(3);
     /// assert_eq!("42000", dec.to_string());
     /// ```
-    pub fn multiply_pow10(&mut self, delta: i16) -> Result<(), Error> {
+    pub fn multiply_pow10(&mut self, delta: i16) {
         match delta.cmp(&0) {
             Ordering::Greater => {
-                self.upper_magnitude = self
-                    .upper_magnitude
-                    .checked_add(delta)
-                    .ok_or(Error::Limit)?;
-                // If we get here, then the magnitude change is in-bounds.
-                let lower_magnitude = self.lower_magnitude + delta;
-                self.lower_magnitude = cmp::min(0, lower_magnitude);
+                let upper_magnitude = self.upper_magnitude.checked_add(delta);
+                match upper_magnitude {
+                    Some(upper_magnitude) => {
+                        self.upper_magnitude = upper_magnitude;
+                        // If we get here, then the magnitude change is in-bounds.
+                        let lower_magnitude = self.lower_magnitude + delta;
+                        self.lower_magnitude = cmp::min(0, lower_magnitude);
+                    }
+                    None => {
+                        // there is an overflow
+                        self.clear();
+                    }
+                }
             }
             Ordering::Less => {
-                self.lower_magnitude = self
-                    .lower_magnitude
-                    .checked_add(delta)
-                    .ok_or(Error::Limit)?;
-                // If we get here, then the magnitude change is in-bounds.
-                let upper_magnitude = self.upper_magnitude + delta;
-                self.upper_magnitude = cmp::max(0, upper_magnitude);
+                let lower_magnitude = self.lower_magnitude.checked_add(delta);
+                match lower_magnitude {
+                    Some(lower_magnitude) => {
+                        self.lower_magnitude = lower_magnitude;
+                        // If we get here, then the magnitude change is in-bounds.
+                        let upper_magnitude = self.upper_magnitude + delta;
+                        self.upper_magnitude = cmp::max(0, upper_magnitude);
+                    }
+                    None => {
+                        // there is an overflow
+                        self.clear();
```

Comment:
And down here, `102030.00` out of range should produce something like `0.00000000...00102030`

Reply:
Ditto, the same comment.